x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -2251,3 +2232,25 @@ function deepMergeVars(original: any, override: any, keepOriginalValue = false):
 
   return result;
 }
+
+async function requireUniqueName(
+  soClient: SavedObjectsClientContract,
+  packagePolicy: UpdatePackagePolicy | NewPackagePolicy,
+  id?: string
+) {
+  const existingPoliciesWithName = await packagePolicyService.list(soClient, {
+    perPage: SO_SEARCH_LIMIT,
+    kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.name:"${packagePolicy.name}"`,
```

Comment:
nit: This logic could be simplified to use ES kuery to filter out the same id, and could use the total count without any hits. 
Probably not a huge gain as there will be at least 1 matches, just some code simplification.

Something like:
```
  const existingPoliciesWithName = await packagePolicyService.list(soClient, {
    perPage: 0,
    kuery: `${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.name:"${packagePolicy.name}" AND NOT (${PACKAGE_POLICY_SAVED_OBJECT_TYPE}.id:"${packagePolicy.id}")`,
  });
if (existingPoliciesWithName?.total > 0) {
  throw ...
}
```

Reply:
Thanks Julia! Since the gain it's not very big I'm going to merge as it is, but we can do this change the next time we touch this area :)